src/actions/commands/actions.ts

```diff
@@ -3158,7 +3158,8 @@ class ActionJoin extends BaseCommand {
           manuallySetCursorPositions: true,
         });
 
-        vimState.cursorStartPosition = vimState.cursorStopPosition = startPosition.withColumn(
```

Comment:
Could you explain what's changing here?

Reply:
In `withColumn`, the length is rounded to the length of the original text in the `startPosition`, so the value of `trimmedLinesContent.length - columnDeltaOffset` is not used correctly.